src/view/main_scene_view.cpp

```diff
@@ -3,18 +3,79 @@
 #include "view/main_scene_view.hpp"
 
 MainSceneView::MainSceneView()
-: camera(glm::radians(45.0f), 800 / 600, .1f, 100.0f)
+: View()
 {
-	//todo temporary below
-	camera.setPosition(glm::vec3(.0f, 3.0f, 10.0f));
+	camera = new AboveCamera(glm::radians(45.0f), 800 / 600, .1f, 100.0f, 0.05f);
+	camera->setPosition(glm::vec3(0.0f, 3.0f, 0.0f));
+
+}
+
+void MainSceneView::processEvents(SDL_Event &event)
+{
+	AboveCamera* aboveCamera = (AboveCamera*)camera;
+
+	switch(event.type) {
+		case SDL_KEYDOWN:
+			switch(event.key.keysym.sym){
+				case SDLK_a:
+					aboveCamera->setMovingLeft(true);
```

Comment:
Ok, so as per my above comment, all methods here have to have interfaces in the Camera base class so that we can use polymorphism. Please use the 
    void moveForward() override; and similar methods from camera here.